### PR TITLE
Create a line chart component to show Experiment Report

### LINKED ISSUE #24331: [UI] Implement Line chart for Reports
### Parent Issue

https://github.com/dotCMS/core/issues/23625

### User Story

As a user, I want to be able to see a report of a selected experiment, 

### Acceptance Criteria

- When going to Experiment Report, see a Line Chart component working
- Implemented design from Figma

### Proposed Objective

User Experience

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
Screenshot 2023-03-15 at 9 10 42 AM

### Assumptions & Initiation Needs

Use https://primeng.org/chart 

### Quality Assurance Notes & Workarounds

_No response_

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Create a component to implement the  design  of Figma of line chart report to use in Experiment Reports

### Checklist
- [X] Tests
- [X] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
Mock data used

### Screenshots
**Design**

Screenshot 2023-03-16 at 4 27 34 PM

**Implementation**
Mar-16-2023 16-26-07

### CODE REVIEW

**Comment by oidacra** [comment]
Waiting for merge of https://github.com/dotCMS/core/pull/24345

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    templateUrl: './dot-experiments-reports-chart.component.html',
+    styleUrls: ['./dot-experiments-reports-chart.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotExperimentsReportsChartComponent {
+    readonly customChartPlugins = [htmlLegendPlugin];
+    readonly dotExperimentsChartJSOptions = dotExperimentChartJsOptions;
+
+    labels = this.addWeekdayToDateLabels([
+        '03/15/2023',
```
Comment (alfredo-dotcms) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.ts:56]: is this ok? or it's WIP?

Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.ts:56]: Waiting for the API part.

File: core-web/libs/dotcms-models/src/lib/dot-experiments-constants.ts [hunk 2]
```diff
@@ -99,3 +99,13 @@ export const GOALS_METADATA_MAP: Record<GOAL_TYPES, { label: string; description
         description: 'experiments.goal.click_on_element.description'
     }
 };
+
+export const daysOfTheWeek = [
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/libs/dotcms-models/src/lib/dot-experiments-constants.ts:103]: With I translated all at `DotExperimentsReportsChartComponent`
```
private addWeekdayToDateLabels(labels: Array<string>) {
        return labels.map((item) => {
            const date = new Date(item).getDay();

            return [this.dotMessageService.get(daysOfTheWeek[date]), item];
        });
    }
```

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-list/dot-experiments-list.component.ts [hunk 3]
```diff
@@ -97,6 +97,23 @@ export class DotExperimentsListComponent {
...
+     */
+    goToViewExperimentReport(experiment: DotExperiment) {
+        this.router.navigate(['/edit-page/experiments/reports/', experiment.id], {
+            queryParams: {
+                editPageTab: null,
+                variantName: null,
+                experimentId: null
+            },
+            queryParamsHandling: 'merge'
+        });
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-list/dot-experiments-list.component.ts:114]: This should be a link and not a click button, when an href change is a navigation which means a link.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/options/dotExperiments-chartjs.options.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            callbacks: {
+                title: function (context) {
+                    const [, title] = context[0].label.split(',');
+
+                    return title;
+                },
+                label: function (context) {
+                    const label = context.dataset.label || '';
+
+                    return ` ${label}: ${context.parsed.y}`;
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/options/dotExperiments-chartjs.options.ts:23]:
```suggestion
                    return `${label}: ${context.parsed.y}`;
```

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/options/dotExperiments-chartjs.options.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    interaction: {
+        mode: 'index',
+        intersect: false
+    },
+
+    scales: {
+        x: {
+            title: {
+                display: true,
+                text: 'Days',
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/options/dotExperiments-chartjs.options.ts:48]: Translation?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/plugins/dotHtmlLegend-chartjs.plugin.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            textContainer.style.padding = '0';
+
+            const text = document.createTextNode(item.text);
+            textContainer.appendChild(text);
+
+            li.appendChild(textContainer);
+            ul.appendChild(li);
+        });
+    }
+};
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/plugins/dotHtmlLegend-chartjs.plugin.ts:66]: Nice!

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.ts [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    VARIANT1: {
+        borderColor: 'rgba(177,117,255,1)',
+        pointBackgroundColor: 'rgba(177,117,255,1)',
+        backgroundColor: 'rgba(177,117,255,.1)'
+    },
+    VARIANT2: {
+        borderColor: 'rgba(65,219,247,1)',
+        pointBackgroundColor: 'rgba(65,219,247,1)',
+        backgroundColor: 'rgba(65,219,247,.1)'
+    }
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.ts:40]: Create constants for this colors.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/dot-experiments-reports.component.scss [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+:host {
+    width: 100%;
+    height: 100%;
+    display: flex;
+    flex-direction: column;
+    padding-bottom: $spacing-7;
+}
+
+.dot-experiment-report-content {
+    padding: $spacing-3 $spacing-5 $spacing-5 $spacing-5;
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/dot-experiments-reports.component.scss:12]:
```suggestion
    padding: $spacing-3 $spacing-5 $spacing-5;
```

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/dot-experiments-reports.component.ts [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    /**
+     * Go to Experiment List
+     * @param {string} pageId
+     * @returns void
+     * @memberof DotExperimentsReportsComponent
+     */
+    goToExperimentList(pageId: string) {
+        this.router.navigate(['/edit-page/experiments/', pageId], {
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/dot-experiments-reports.component.ts:62]: Link.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-shell/dot-experiments-shell-routing.module.ts [hunk 10]
```diff
@@ -29,6 +29,13 @@ const routes: Routes = [
                             '../dot-experiments-configuration/dot-experiments-configuration.module'
                         )
                     ).DotExperimentsConfigurationModule
+            },
+            {
+                path: 'reports',
+                title: 'Experiment Reports',
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-shell/dot-experiments-shell-routing.module.ts:35]: This needs translation.

File: core-web/libs/dot-primeng-theme-styles/src/scss/dotcms-theme/components/_tag.scss [hunk 11]
```diff
@@ -12,27 +12,56 @@
...
+        background-color: $color-accessible-text-purple-bg;
+    }
+
+    &.ended {
+        color: $color-accessible-text-blue;
+        background-color: $color-accessible-text-blue-bg;
+    }
+
+    &.archived {
+        color: $color-accessible-text-gray;
```
Comment (fmontes) [inline core-web/libs/dot-primeng-theme-styles/src/scss/dotcms-theme/components/_tag.scss:64]: This is only for experiments not all the tags component for that should not be in this file.

File: core-web/libs/dot-primeng-theme-styles/src/scss/_variables.scss [hunk 12]
```diff
@@ -219,3 +220,6 @@ $tabView-scroll-btn-width: 40px;
 
 // Tag
 $tag-radius-md: 7px;
+
+// Experiments
+$container-reports-max: 1100px;
```
Comment (fmontes) [inline core-web/libs/dot-primeng-theme-styles/src/scss/_variables.scss:225]: Move this to it own context, this is not a global var. I am fixing some of this in the unify SCSS vars PR

File: core-web/package.json [hunk 13]
```diff
@@ -87,10 +87,10 @@
     "@tiptap/starter-kit": "^2.0.0-beta.218",
     "@tiptap/suggestion": "^2.0.0-beta.218",
     "@webcomponents/webcomponentsjs": "^2.6.0",
-    "chart.js": "^3.3.2",
+    "chart.js": "^4.2.1",
```
Comment (fmontes) [inline core-web/package.json:90]: Did you check dotCDN?

**Review by fmontes — DISMISSED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-list/dot-experiments-list.component.ts (hunk 3 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-list/dot-experiments-list.component.ts:114]: That method comes from an output of a child component(`DotExperimentsListTableComponent`).
`(goToReport)="goToViewExperimentReport($event)"`

File: core-web/libs/dot-primeng-theme-styles/src/scss/_variables.scss (hunk 12 above)
Comment (oidacra) [reply core-web/libs/dot-primeng-theme-styles/src/scss/_variables.scss:225]: moved

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/options/dotExperiments-chartjs.options.ts (hunk 4 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/options/dotExperiments-chartjs.options.ts:23]: Done.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/options/dotExperiments-chartjs.options.ts (hunk 5 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/options/dotExperiments-chartjs.options.ts:48]: Done.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.ts (hunk 7 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/dot-experiments-reports-chart.component.ts:40]: done!

File: core-web/package.json (hunk 13 above)
Comment (oidacra) [reply core-web/package.json:91]: I checked the config, and it does not show any problem at the IDE, but I can't find where or how to run that chart.

File: core-web/libs/dot-primeng-theme-styles/src/scss/dotcms-theme/components/_tag.scss
Comment (alfredo-dotcms) [inline core-web/libs/dot-primeng-theme-styles/src/scss/dotcms-theme/components/_tag.scss:1]: delete?

Comment (oidacra) [reply core-web/libs/dot-primeng-theme-styles/src/scss/dotcms-theme/components/_tag.scss:1]: Comes from the refactor of scss. Now in master that file has other place.

**Review by alfredo-dotcms — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/options/dotExperiments-chartjs.options.ts [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+                        size: 15,
+                        lineHeight: 1.2
+                    },
+                    padding: 5
+                },
+                ticks: {
+                    callback: function (val, index) {
+                        return index % 2 === 0 ? this.getLabelForValue(val as number) : '';
+                    },
+                    color: '#524E5C',
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/options/dotExperiments-chartjs.options.ts:68]: That property only accept strings. A hex or rgb color.

File: core-web/libs/dot-primeng-theme-styles/src/scss/_variables.scss
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/libs/dot-primeng-theme-styles/src/scss/_variables.scss:1]: Comes from the recent refactoring of the scss.

**Review by hmoreras — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/options/dotExperiments-chartjs.options.ts (hunk 14 above)
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/components/dot-experiments-reports-chart/chartjs/options/dotExperiments-chartjs.options.ts]: Done!

**Pull request merged**